src/routes/CourseDetail.tsx

```diff
@@ -358,22 +371,34 @@ const CourseDetail = (props: Props) => {
     return !(!course.subcourse || isStudent);
   };
 
-  const hasEnded = () => {
+  const getLastLectureEnd = () => {
     const lectures = course.subcourse.lectures.sort(
       (a, b) => a.start - b.start
     );
     const lastLecture = lectures[lectures.length - 1];
     if (lastLecture != null) {
-      const lectureEnd = moment
+      return moment
         .unix(lastLecture.start)
         .add(lastLecture.duration, 'minutes');
+    }
+    return null;
+  };
 
-      return moment().isAfter(lectureEnd);
+  const hasEnded = () => {
+    if (getLastLectureEnd) {
+      return moment().isAfter(getLastLectureEnd());
     }
 
     return false;
   };
 
+  const canContact = () => {
+    if (getLastLectureEnd) {
```

Comment:
ts ts ts ... who reviewed this ;) ... https://github.com/corona-school/web-user-app/commit/af01ff6470341db6cf0175e1a8536bb2b9ccf98a